sentry-android-core/src/main/java/io/sentry/android/core/SentryPerformanceProvider.java

```diff
@@ -98,6 +99,11 @@ public void shutdown() {
       if (appStartProfiler != null) {
         appStartProfiler.close();
       }
+      final @Nullable IContinuousProfiler appStartContinuousProfiler =
```

Comment:
as I understood there could be only one profiler at a time (cont. or tx-based), right? Just got an idea we could have a general `IProfiler` interface (that would have all the methods that `IContinuousProfiler` currently has), and then it'd have a sub-interface `ITransactionProfiler` with the tx-specific methods.

This way you could only manage one instance of `IProfiler` everywhere (like here) and it would be more readable I guess.

Reply:
you're probably right on this, but in the long term we will deprecate and optionally remove the transaction profiler.
I think the extraction of the `AndroidProfiler` is enough for now.
But we can refactor it if you have a strong preference